src/main/java/slimeknights/tconstruct/smeltery/recipe/AbstractCastingRecipe.java

```diff
@@ -0,0 +1,202 @@
+package slimeknights.tconstruct.smeltery.recipe;
```

Comment:
Why is this not in library? Recipes are part of our "public" API as other mods will likely use them (in the builder that is)

Reply:
because I just picked a spot to use so I didn't get stuck in 'what's the right place for this'
mode. Just the AbstractCastingRecipe, or its two children?

Reply:
Yeah, I know the packages are a little weird. Eventually I will make a proper guide of what goes in library and what does not.

For the non-abstract recipes, it a good question. Officially, mods should not need to use the class, directly but on the other hand, it makes it all in one place. My rule of thumb before we had JSON recipes was if its a special case recipe, keep it in the module, but if its a recipe other mods may use, put it in library. I guess we can stick with that for the modules.

Reply:
On another note, are we going to perhaps make an api jar for folks to use?
I'm not certain as to the mechanics of that in modding context.

Reply:
Its definitely not possible with our current library structure. Its something we can look into in the future, but not a priority.